modules/InboundEmail/InboundEmail.php

```diff
@@ -5376,34 +5375,15 @@ public function returnImportedEmail($msgNo, $uid, $forDisplay = false, $clean_em
                 // Store CIDs in imported messages, convert on display
                 $this->imagePrefix = 'cid:';
             }
-            // handle multi-part email bodies
-            $subtypeArray = [
-                'MIXED',
-                'ALTERNATIVE',
-                'RELATED',
-                'REPORT',
-                'HTML'
-            ];
-
-            if (in_array(strtoupper($structure->subtype), $subtypeArray, true)) {
-                $email->description_html = $this->getMessageTextWithUid(
-                    $uid,
-                    $structure->subtype,
-                    $structure,
-                    $fullHeader,
-                    $clean_email
-                );
-            } elseif ($structure->subtype === 'PLAIN') {
-                $email->description = $this->getMessageTextWithUid(
-                    $uid,
-                    'PLAIN',
-                    $structure,
-                    $fullHeader,
-                    $clean_email
-                );
-            } else {
-                LoggerManager::getLogger()->warn('Unknown MIME subtype in fetch request');
-            }
+
+            $email->description_html = $this->getMessageTextWithUid(
+                $uid,
+                $structure->subtype,
+                $structure,
+                $fullHeader,
+                $clean_email
+            );
+
```

Comment:
Yes, with this implementation we don't need to specify a type or deal with encoding before attempting to parse the email, it will all be handled correctly by the MailMimeParser (Original implementation was actually only really capable of handling explicitly "PLAINTEXT" and "HTML" emails).

Reply:
but  in suite it uses different fields for html and plain test emails description for plain text, description_html for html

Reply:
Ah, I think I see what you mean, the ```email->description``` stuff being removed? I don't really see much point in keeping it tbh, we really shouldn't be trying to parse imap headers for a type before throwing our best guess at ```getMessageTextWithUid``` and hoping it can handle it.

From general regression testing in emails, cases, etc, I was able to get plaintext and non-plantext + complex email types to display correctly.

Reply:
For extra context, the email body itself uses description_html regardless of what type is used here. If PLAINTEXT, then the ```description_html``` will just be set to ```email->description```

```php
            if (empty($email->description_html)) {
                $email->description_html = $email->description;
                $email->description_html = nl2br($email->description_html);
            }
```

Reply:
Tests show this changes how the email is stored, current functionality stores the plain text and the html email, so this would change that functionality, which may have unknown side effects. So we should maintain the current functionality, that supports plain text and html. 
Additionally storing the plain text in the html fields may have unknown side effects as it is. Other than this it work as expected.

Reply:
Agreed, with this change we'd stop using the ```$email->description``` which while fine for core, I could see how this might impact someone with customisations that expects it to have a value. Will update this.